qcodes/instrument_drivers/tektronix/Keithley_3706A_Matrix_channels.py

```diff
@@ -0,0 +1,709 @@
+                               'ranges, slots, or "allslots".')
+        delay_times = [float(x) for x in self.ask(
+                       f"channel.getdelay('{val}')").split(',')]
+        return delay_times
+
+    def set_backplane(self, val: str, backplane: str) -> None:
+        """
+        Sets the analog backplane relays to use with given channels
+        when they are used in switching applications.
+
+        Args:
+            val: A string representing the list of channels to change.
+            backplane: A string representing the list of analog backplane
+                relays to set for the channels specified.
+        """
+        states = self.get_interlock_state()
+        backplanes = self.get_analog_backplane_specifiers()
+        plane_specifiers = backplane.split(',')
+        val_specifiers = val.split(',')
+        for element in states:
+            if element['state'] == 'Interlock is disengaged':
+                warnings.warn(f'The hardware interlock in Slot '
+                              f'{element["slot_no"]} is disengaged. '
+                              'The corresponding analog backplane relays '
+                              'cannot be energized.', UserWarning, 2)
+        for elem in val_specifiers:
+            if elem in backplanes:
+                raise InvalidValue(f'{val} is not a valid specifier. '
+                                   'The specifier cannot be analog '
+                                   'backplane relay.')
+        if not self._validator(val):
+            raise InvalidValue(f'{val} is not a valid specifier. '
+                               'The specifier should be channels, channel '
+                               'ranges, slots, or "allslots".')
+        for plane in plane_specifiers:
+            if plane not in backplanes:
+                raise InvalidValue(f'{backplane} is not a valid specifier. '
+                                   'The specifier should be analog '
+                                   'backplane relay.')
+        self.write(f"channel.setbackplane('{val}', '{backplane}')")
+
+    def get_backplane(self, val: str) -> str:
+        """
+        Lists the backplane relays that are controlled in switching
+        applications for specified channels.
+
+        Args:
+            val: A string representing the channels being queried.
+        """
+        backplanes = self.get_analog_backplane_specifiers()
+        specifiers = val.split(',')
+        for element in specifiers:
+            if element in backplanes:
+                raise InvalidValue(f'{val} cannot be a analog backplane relay.')
+        if not self._validator(val):
+            raise InvalidValue(f'{val} is not a valid specifier. '
+                               'The specifier should be channels, channel '
+                               'ranges, slots, or "allslots".')
+        return self.ask(f"channel.getbackplane('{val}')")
+
+    def _get_slot_id(self) -> List[str]:
```

Comment:
shall it be plural because it returns a list of ids?
```suggestion
    def _get_slot_ids(self) -> List[str]:
```